sql/columndefault.go

```diff
@@ -0,0 +1,141 @@
+package sql
+
+import (
+	"fmt"
+	"strings"
+)
+
+type columnDefaultManager struct {
+	AllowedDefaultFunctions FunctionRegistry
+	ExpressionTransformUp func(Expression) (Expression, error)
+	ParseDefaultString func(*Context, string) (Expression, error)
+}
+var ColumnDefaultManager = &columnDefaultManager{
+	AllowedDefaultFunctions: NewFunctionRegistry(),
+}
+
+// StringToColumnDefaultValue takes in a string representing a default value and returns the equivalent Expression.
+func StringToColumnDefaultValue(ctx *Context, exprStr string) (*ColumnDefaultValue, error) {
```

Comment:
Why doesn't this need to live here, rather than in the parse package?

Reply:
I'm keeping it by the `ColumnDefaultValue` type, as it's the primary way to instantiate it as far as integrators are concerned. It appears that the `sql` package is the place for most of the interfaces and types for integrators to add resides, so this follows that trend.

I remember running into an import cycle with having all of this done in the `parse` package before, but I've rewritten quite a lot since then, so perhaps it might not happen again. I'll check.

Reply:
This has been moved to the `parse` package in the latest iteration. Still not a fan of the instantiation of a type not being right beside the type.